providers/README.md

```diff
@@ -0,0 +1,90 @@
+# Main Checkbox providers reviewer guidance
+
+In the interest of making the code uniform and maintainable for at least the
+Extended Maintenance Period (ESM) that the tests in the main providers usually
+outlive, the following describes the code style we follow. There are examples
+of deviation from this style in the main providers, but they either have a
+motivation or are candidates for refactoring.
+
+This document is not a replacement for automated linting, it is a supplement.
+
+You can violate the following rules but when you do so we expect a strong
+motivation to be expressed either in the PR or in comments in the code itself.
+
+# Dependencies
+
+Dependencies are very painful to manage in Checkbox because we support every
+Ubuntu version still under ESM. Avoid using dependencies at all costs.
```

Comment:
```suggestion
Dependencies are very painful to manage in Checkbox because every Ubuntu
version still under ESM must be supported. Avoid using dependencies at
all costs.
```